Rx/v2/src/rxcpp/operators/rx-amb.hpp

```diff
@@ -172,35 +216,88 @@ struct amb
     }
 };
 
-template<class Coordination>
-class amb_factory
+}
+
+/*! @copydoc rx-amb.hpp
+*/
+template<class... AN>
+auto amb(AN&&... an)
+    ->     operator_factory<amb_tag, AN...> {
+    return operator_factory<amb_tag, AN...>(std::make_tuple(std::forward<AN>(an)...));
+}
+
+}
+
+template<class SourceObservable, class Observable, class Coordination>
+struct defer_amb : public defer_observable<
```

Comment:
are these defer constructs still required? I was hoping that the new pattern would not require these.